specification/postgresql/Microsoft.OrionDB.Management/models.tsp

```diff
@@ -0,0 +1,547 @@
+import "@azure-tools/typespec-azure-resource-manager";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using Versioning;
+using Azure.ResourceManager;
+using Azure.ResourceManager.Foundations;
+
+@armProviderNamespace("Microsoft.OrionDB")
+namespace Microsoft.OrionDB;
+
+/**
+ * The PG engine majorversion for OrionDb cluster.
+ */
+union OrionDbVersion {
+  string,
+  /** PG engine majorversion 17 */
+  Seventeen: "17",
+  /** PG engine majorversion 16 */
+  Sixteen: "16",
+  /** PG engine majorversion 15 */
+  Fifteen: "15",
+  /** PG engine majorversion 14 */
+  Fourteen: "14",
+  /** PG engine majorversion 13 */
+  Thirteen: "13",
+}
+
+/**
+ * A state of a Orion Db cluster that is visible to user.
+ */
+union ClusterState {
+  string,
+  /** Represents it is created and ready to use. */
+  Ready: "Ready",
+  /** Represents it is being dropped. */
+  Dropping: "Dropping",
+  /** Represents it is disabled. */
+  Disabled: "Disabled",
+  /** Represents it is starting. */
+  Starting: "Starting",
+  /** Represents it is stopping. */
+  Stopping: "Stopping",
+  /** Represents it is stopped. */
+  Stopped: "Stopped",
+  /** Represents it is being updated. */
+  Updating: "Updating",
+}
+
+/**
+ * The mode to create a new Orion Db cluster
+ */
+union CreateMode {
+  string,
+  /** Create or Update a new Orion Db cluster */
+  Default: "Default",
+  /** Create a new Orion Db cluster */
+  Create: "Create",
+  /** Update an existing Orion Db cluster. */
+  Update: "Update",
+  /** Restore an existing Orion Db cluster to a point in time. */
+  PointInTimeRestore: "PointInTimeRestore",
+  /** Restore an existing Orion Db cluster to a geo location. */
+  GeoRestore: "GeoRestore",
+  /** Revive a dropped Orion Db cluster. */
+  ReviveDropped: "ReviveDropped",
+}
+
+/**
+ * Flag to enable / disable Storage Auto grow for an Orion Db cluster.
+ */
+union StorageAutoGrow {
+  string,
+  /** Enable Storage Auto grow for an Orion Db cluster. */
+  Enabled: "Enabled",
+  /** Disable Storage Auto grow for an Orion Db cluster. */
+  Disabled: "Disabled",
+}
+
+/**
+ * Name of storage tier for IOPS.
+ */
+union AzureManagedDiskPerformanceTiers {
+  string,
+  /** Performance tier P1 */
+  P1: "P1",
+  /** Performance tier P2 */  
+  P2: "P2",
+  /** Performance tier P3 */
+  P3: "P3",
+  /** Performance tier P4 */
+  P4: "P4",
+  /** Performance tier P6 */
+  P6: "P6",
+  /** Performance tier P10 */
+  P10: "P10",
+  /** Performance tier P15 */
+  P15: "P15",
+  /** Performance tier P20 */
+  P20: "P20",
+  /** Performance tier P30 */
+  P30: "P30",
+  /** Performance tier P40 */
+  P40: "P40",
+  /** Performance tier P50 */
+  P50: "P50",
+  /** Performance tier P60 */
+  P60: "P60",
+  /** Performance tier P70 */
+  P70: "P70",
+  /** Performance tier P80 */
+  P80: "P80",
+}
+
+/**
+ * Storage type for the cluster. Allowed values are Premium_LRS, PremiumV2_LRS, and UltraSSD_LRS. Default is Premium_LRS if not specified
+ */
+union StorageType {
```

Comment:
I don't think customer can select storage type for the only managed disk that a replica has attached to store configuration files.

Reply:
removed